### PR TITLE
Adding network-Stress cases and support for two host configuration for ltp tests

### PR DESCRIPTION
**Pull request by TasmiyaNalatwad**
The PR is to support ltp test between 2 hosts and adding network stress bucket.
1. The code had input parameter called two-host-configuration passed into yaml file. If the parameter is set to "True" Ltp runs between 2 hosts, otherwise default value is "False" and ltp runs on single machine
2. Necessary services restart is added.
3. Ltp Network-Stress bucket is added.

### CODE REVIEW

**Comment by TasmiyaNalatwad** [comment]
@abdhaleegit Please find the results below for two-host-configuration
I have rerun the test cases which were failing due to compilation failing in Peer machine. Few cases which are failed says as expected in test results. May be those are having any issue while running ltp between two-host machines. I tried several times to run thoses cases even manually, those cases were failing everytime.

[root@ltcden11-lp8 net]# cat nohup.out 
Fetching asset from ltp_net.py:LTP.test
Fetching asset from ltp_net.py:LTP.test
JOB ID     : 424307585000a1155faa081a719dff90327121d1
JOB LOG    : /home/ioci/avocado-fvt-wrapper/results/job-2023-09-13T04.01-4243075/job.log
 (01/20) ltp_net.py:LTP.test;run-runltp-net.features-e75a: STARTED
 (01/20) ltp_net.py:LTP.test;run-runltp-net.features-e75a:  FAIL: LTP tests failed: ['macsec01'] (721.27 s)
 (02/20) ltp_net.py:LTP.test;run-runltp-net.ipv6-be37: STARTED
 (02/20) ltp_net.py:LTP.test;run-runltp-net.ipv6-be37:  CANCEL: Unable to compile ltp in peer machine (173.93 s)
 (03/20) ltp_net.py:LTP.test;run-runltp-net.ipv6_lib-34ab: STARTED
 (03/20) ltp_net.py:LTP.test;run-runltp-net.ipv6_lib-34ab:  PASS (244.13 s)
 (04/20) ltp_net.py:LTP.test;run-runltp-net.nfs-35ed: STARTED
 (04/20) ltp_net.py:LTP.test;run-runltp-net.nfs-35ed:  FAIL: LTP tests failed: ['nfs4_08', 'nfs41_08', 'nfs42_08', 'nfs4_ipv6_08', 'nfs41_ipv6_08', 'nfs42_ipv6_08', 'nfslock3t_01', 'nfslock3t_ipv6_01'] (3599.60 s)
 (05/20) ltp_net.py:LTP.test;run-runltp-net.tcp_cmds-6253: STARTED

---------------------------------------------------------------------------------------------------------------------------------
Fetching asset from ltp_net.py:LTP.test
Fetching asset from ltp_net.py:LTP.test
JOB ID     : 02834c1f182018b802209ab05490040e783ac98a
JOB LOG    : /home/ioci/avocado-fvt-wrapper/results/job-2023-09-13T05.38-02834c1/job.log
 (01/15) ltp_net.py:LTP.test;run-runltp-net.ipv6-be37: STARTED
 (01/15) ltp_net.py:LTP.test;run-runltp-net.ipv6-be37:  FAIL: LTP tests failed: ['tracepath601'] (257.29 s)
 (02/15) ltp_net.py:LTP.test;run-runltp-net.sctp-f4fd: STARTED
 (02/15) ltp_net.py:LTP.test;run-runltp-net.sctp-f4fd:  PASS (248.64 s)
 (03/15) ltp_net.py:LTP.test;run-runltp-net.multicast-e0a1: STARTED
 (03/15) ltp_net.py:LTP.test;run-runltp-net.multicast-e0a1:  PASS (1054.91 s)
 (04/15) ltp_net.py:LTP.test;run-runltp-net.rpc_tests-7ba5: STARTED
 (04/15) ltp_net.py:LTP.test;run-runltp-net.rpc_tests-7ba5:  FAIL: LTP tests failed: ['rpc01', 'rpcinfo', 'rpc_pmap_set', 'rpc_pmap_unset', 'rpc_pmap_getport', 'rpc_pmap_getmaps', 'rpc_pmap_rmtcall', 'rpc_get_myaddress', 'rpc_authnone_create', 'rpc_authunix_create', 'rpc_authunix_create_default', 'rpc_auth_destroy', 'rpc... (242.10 s)
 (05/15) ltp_net.py:LTP.test;run-runltp-net.tirpc_tests-236e: STARTED
 (05/15) ltp_net.py:LTP.test;run-runltp-net.tirpc_tests-236e:  FAIL: LTP tests failed: ['tirpc_rpcb_getaddr', 'tirpc_rpcb_getmaps', 'tirpc_authnone_create', 'tirpc_authsys_create', 'tirpc_authsys_create_default', 'tirpc_clnt_dg_create', 'tirpc_svc_dg_create', 'tirpc_clnt_vc_create', 'tirpc_svc_vc_create', 'tirpc_bottomleve... (238.65 s)
 (06/15) ltp_net.py:LTP.test;run-runltp-net_stress.appl-905d: STARTED
 (06/15) ltp_net.py:LTP.test;run-runltp-net_stress.appl-905d:  CANCEL: Unable to compile ltp in peer machine (172.59 s)
 (07/15) ltp_net.py:LTP.test;run-runltp-net_stress.broken_ip-1f35: STARTED
 (07/15) ltp_net.py:LTP.test;run-runltp-net_stress.broken_ip-1f35:  PASS (407.89 s)
 (08/15) ltp_net.py:LTP.test;run-runltp-net_stress.interface-6435: STARTED
 (08/15) ltp_net.py:LTP.test;run-runltp-net_stress.interface-6435:  PASS (367.07 s)
 (09/15) ltp_net.py:LTP.test;run-runltp-net_stress.ipsec_dccp-de7d: STARTED
 (09/15) ltp_net.py:LTP.test;run-runltp-net_stress.ipsec_dccp-de7d:  PASS (287.23 s)
 (10/15) ltp_net.py:LTP.test;run-runltp-net_stress.ipsec_icmp-dd11: STARTED
 (10/15) ltp_net.py:LTP.test;run-runltp-net_stress.ipsec_icmp-dd11:  PASS (478.59 s)
 (11/15) ltp_net.py:LTP.test;run-runltp-net_stress.ipsec_sctp-4567: STARTED
 (11/15) ltp_net.py:LTP.test;run-runltp-net_stress.ipsec_sctp-4567:  CANCEL: Unable to compile ltp in peer machine (175.42 s)
 (12/15) ltp_net.py:LTP.test;run-runltp-net_stress.ipsec_tcp-1bd0: STARTED
 (12/15) ltp_net.py:LTP.test;run-runltp-net_stress.ipsec_tcp-1bd0:  PASS (701.45 s)
 (13/15) ltp_net.py:LTP.test;run-runltp-net_stress.ipsec_udp-ea25: STARTED
 (13/15) ltp_net.py:LTP.test;run-runltp-net_stress.ipsec_udp-ea25:  PASS (1023.76 s)
 (14/15) ltp_net.py:LTP.test;run-runltp-net_stress.multicast-6665: STARTED
 (14/15) ltp_net.py:LTP.test;run-runltp-net_stress.multicast-6665:  PASS (425.38 s)
 (15/15) ltp_net.py:LTP.test;run-runltp-net_stress.route-6760: STARTED
 (15/15) ltp_net.py:LTP.test;run-runltp-net_stress.route-6760:  PASS (264.54 s)
RESULTS    : PASS 10 | ERROR 0 | FAIL 3 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 2
JOB HTML   : /home/ioci/avocado-fvt-wrapper/results/job-2023-09-13T05.38-02834c1/results.html
JOB TIME   : 6495.07 s

------------------------------------------------------------------------------------------
Fetching asset from ltp_net.py:LTP.test
Fetching asset from ltp_net.py:LTP.test
JOB ID     : 65875d94b6143116d996895950a24f6ba9f1a72a
JOB LOG    : /home/ioci/avocado-fvt-wrapper/results/job-2023-09-13T09.37-65875d9/job.log
 (1/2) ltp_net.py:LTP.test;run-runltp-net_stress.appl-905d: STARTED
 (1/2) ltp_net.py:LTP.test;run-runltp-net_stress.appl-905d:  PASS (329.32 s)
 (2/2) ltp_net.py:LTP.test;run-runltp-net_stress.ipsec_sctp-4567: STARTED
 (2/2) ltp_net.py:LTP.test;run-runltp-net_stress.ipsec_sctp-4567:  PASS (279.38 s)
RESULTS    : PASS 2 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/ioci/avocado-fvt-wrapper/results/job-2023-09-13T09.37-65875d9/results.html
JOB TIME   : 648.36 s

File: io/net/ltp_net.py [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2016 IBM
+# Author: Santhosh G <santhog4@linux.vnet.ibm.com>
```
Comment (abdhaleegit) [inline io/net/ltp_net.py:15]: Please update author and date

File: io/net/ltp_net.py [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+                              fstype="tmpfs", mnt_check=False)
+
+    def setUp(self):
+        smg = SoftwareManager()
+        dist = distro.detect()
+        self.args = self.params.get('args', default='')
+        self.mem_leak = self.params.get('mem_leak', default=0)
+        self.peer_public_ip = self.params.get("peer_public_ip", default="")
+        self.peer_user = self.params.get("peer_user", default="root")
+        self.peer_password = self.params.get("peer_password", '*',
```
Comment (abdhaleegit) [inline io/net/ltp_net.py:86]: remove the path param '*' not needed

**Review by abdhaleegit — CHANGES_REQUESTED** [review]
@TasmiyaNalatwad Please remove the test from yaml which are failing, those can be taken as separate PR

**Review by PraveenPenguin — CHANGES_REQUESTED** [review]
@TasmiyaNalatwad why new test case, can you please check if we able to extend the existing LTP test case to cover this

**Comment by TasmiyaNalatwad** [comment]
> @TasmiyaNalatwad why new test case, can you please check if we able to extend the existing LTP test case to cover this

@PraveenPenguin New LTP is for Network cases, as we are modifying the LTP according to network test bucket, there are many changes done to make network cases run successfully. Like Network cases need some mandatory services to be started,  most of the network cases need 2 hosts to run tests like nfs, tcp, udp etc and all these dependencies are not required for other test buckets. 
Hence making LTP specific to Network test bucket in avocado-misc-test/io/net and keeping the avocado-misc-test/generic LTP untouched.

@abdhaleegit Please confirm on the same.

**Comment by PraveenPenguin** [comment]
> > @TasmiyaNalatwad why new test case, can you please check if we able to extend the existing LTP test case to cover this
> 
> @PraveenPenguin New LTP is for Network cases, as we are modifying the LTP according to network test bucket, there are many changes done to make network cases run successfully. Like Network cases need some mandatory services to be started, most of the network cases need 2 hosts to run tests like nfs, tcp, udp etc and all these dependencies are not required for other test buckets. Hence making LTP specific to Network test bucket in avocado-misc-test/io/net and keeping the avocado-misc-test/generic LTP untouched.
> 
> @abdhaleegit Please confirm on the same.

@TasmiyaNalatwad thanks for the detailed info, i know, what i thinking/suggesting have code which is needed for n/w go with a function and call that if the network test called via yaml and rescue common code , this will reduce of maintenance over head and use case

**Comment by TasmiyaNalatwad** [comment]
@abdhaleegit @PraveenPenguin Updating the latest results with present test cases

JOB ID     : 909a45b0563e9116cd9e5580665bdba7fd3df0fa
JOB LOG    : /home/ioci/avocado-fvt-wrapper/results/job-2023-09-14T07.25-909a45b/job.log

 (02/14) ltp_net.py:LTP.test;run-runltp-net.ipv6_lib-c674: STARTED
 (02/14) ltp_net.py:LTP.test;run-runltp-net.ipv6_lib-c674:  PASS (412.30 s)
 (03/14) ltp_net.py:LTP.test;run-runltp-net.sctp-30b4: STARTED
 (03/14) ltp_net.py:LTP.test;run-runltp-net.sctp-30b4:  PASS (404.41 s)
 (04/14) ltp_net.py:LTP.test;run-runltp-net.multicast-d35d: STARTED
 (04/14) ltp_net.py:LTP.test;run-runltp-net.multicast-d35d:  PASS (1224.59 s)
 (05/14) ltp_net.py:LTP.test;run-runltp-net_stress.appl-d901: STARTED
 (05/14) ltp_net.py:LTP.test;run-runltp-net_stress.appl-d901:  PASS (530.24 s)
 (06/14) ltp_net.py:LTP.test;run-runltp-net_stress.broken_ip-1ac6: STARTED
 (06/14) ltp_net.py:LTP.test;run-runltp-net_stress.broken_ip-1ac6:  PASS (573.99 s)
 (07/14) ltp_net.py:LTP.test;run-runltp-net_stress.interface-95f8: STARTED
 (07/14) ltp_net.py:LTP.test;run-runltp-net_stress.interface-95f8:  PASS (533.79 s)
 (08/14) ltp_net.py:LTP.test;run-runltp-net_stress.ipsec_dccp-64c6: STARTED
 (08/14) ltp_net.py:LTP.test;run-runltp-net_stress.ipsec_dccp-64c6:  PASS (449.88 s)
 (09/14) ltp_net.py:LTP.test;run-runltp-net_stress.ipsec_icmp-947a: STARTED
 (09/14) ltp_net.py:LTP.test;run-runltp-net_stress.ipsec_icmp-947a:  PASS (666.45 s)
 (10/14) ltp_net.py:LTP.test;run-runltp-net_stress.ipsec_sctp-66f9: STARTED
 (10/14) ltp_net.py:LTP.test;run-runltp-net_stress.ipsec_sctp-66f9:  PASS (443.28 s)
 (11/14) ltp_net.py:LTP.test;run-runltp-net_stress.ipsec_tcp-1abd: STARTED
 (11/14) ltp_net.py:LTP.test;run-runltp-net_stress.ipsec_tcp-1abd:  PASS (851.09 s)
 (12/14) ltp_net.py:LTP.test;run-runltp-net_stress.ipsec_udp-bbb7: STARTED
 (12/14) ltp_net.py:LTP.test;run-runltp-net_stress.ipsec_udp-bbb7:  PASS (1241.24 s)
 (13/14) ltp_net.py:LTP.test;run-runltp-net_stress.multicast-2a6a: STARTED
 (13/14) ltp_net.py:LTP.test;run-runltp-net_stress.multicast-2a6a:  PASS (596.83 s)
 (14/14) ltp_net.py:LTP.test;run-runltp-net_stress.route-36b1: STARTED
 (14/14) ltp_net.py:LTP.test;run-runltp-net_stress.route-36b1:  PASS (425.60 s)
RESULTS    : PASS 13 | ERROR 1 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/ioci/avocado-fvt-wrapper/results/job-2023-09-14T07.25-909a45b/results.html
JOB TIME   : 8508.37 s

[root@ltcden11-lp14 net]# avocado run --max-parallel-tasks=1 ltp_net.py -m ltp_net.py.data/ltp_net.yaml
Fetching asset from ltp_net.py:LTP.test
Fetching asset from ltp_net.py:LTP.test
JOB ID     : 1af83c8c292866b5b90d2af471bb600f6be1c38f
JOB LOG    : /root/avocado-fvt-wrapper/results/job-2023-09-15T01.27-1af83c8/job.log
 (1/1) ltp_net.py:LTP.test;run-runltp-net.ipv6-a6a3: STARTED
 (1/1) ltp_net.py:LTP.test;run-runltp-net.ipv6-a6a3: PASS (400.68 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/avocado-fvt-wrapper/results/job-2023-09-15T01.27-1af83c8/results.html
JOB TIME   : 432.65 s

**Comment by TasmiyaNalatwad** [comment]
@PraveenPenguin @abdhaleegit The one more reason having a separate file was, 
In common code LTP is installed and compiled in /root/avocado-fvt-wrapper/results/job_directory/test-results/tmp_file 
For each single test case LTP is installed and compiled and later it is not deleted, bcz of which the device storage is getting full for every network test bucket as the number of test cases are more. LTP itself is of 1.8GB and  debug.log file is going more than 12M bcz of compilation and installation logs getting captured. 
Moreover Network test bucket require LTP to be installed in host and peer lpars. So i am downloading ltp in /home/ltp path and deleting the directory at the end of each case.

Would you please let me know should i go to the old file and modify the code for Network tests. I still have the old PR open i will modify there and resent it again.

**Comment by PraveenPenguin** [comment]
> @PraveenPenguin @abdhaleegit The one more reason having a separate file was, In common code LTP is installed and compiled in /root/avocado-fvt-wrapper/results/job_directory/test-results/tmp_file For each single test case LTP is installed and compiled and later it is not deleted, bcz of which the device storage is getting full for every network test bucket as the number of test cases are more. LTP itself is of 1.8GB and debug.log file is going more than 12M bcz of compilation and installation logs getting captured. Moreover Network test bucket require LTP to be installed in host and peer lpars. So i am downloading ltp in /home/ltp path and deleting the directory at the end of each case.
> 
> Would you please let me know should i go to the old file and modify the code for Network tests. I still have the old PR open i will modify there and resent it again.

@TasmiyaNalatwad nice catch, can you please send a fix to delete old compile files  as this also hits us when we want to copy logs, I still advocate having it in single place, let us wait to hear from @abdhaleegit what he says

**Comment by abdhaleegit** [comment]
> > @PraveenPenguin @abdhaleegit The one more reason having a separate file was, In common code LTP is installed and compiled in /root/avocado-fvt-wrapper/results/job_directory/test-results/tmp_file For each single test case LTP is installed and compiled and later it is not deleted, bcz of which the device storage is getting full for every network test bucket as the number of test cases are more. LTP itself is of 1.8GB and debug.log file is going more than 12M bcz of compilation and installation logs getting captured. Moreover Network test bucket require LTP to be installed in host and peer lpars. So i am downloading ltp in /home/ltp path and deleting the directory at the end of each case.
> > Would you please let me know should i go to the old file and modify the code for Network tests. I still have the old PR open i will modify there and resent it again.
> 
> @TasmiyaNalatwad nice catch, can you please send a fix to delete old compile files as this also hits us when we want to copy logs, I still advocate having it in single place, let us wait to hear from @abdhaleegit what he says

@PraveenPenguin Our initial plan and old PR we had in the same file, but by looking at the upcoming huge design change with LTP network to modify ltp binaries in local and remote host.. so I plan to keep this separate .. else the test will become complex for further enhancenement.  so will keep ltp_network separate

**Comment by PraveenPenguin** [comment]
> > > @PraveenPenguin @abdhaleegit The one more reason having a separate file was, In common code LTP is installed and compiled in /root/avocado-fvt-wrapper/results/job_directory/test-results/tmp_file For each single test case LTP is installed and compiled and later it is not deleted, bcz of which the device storage is getting full for every network test bucket as the number of test cases are more. LTP itself is of 1.8GB and debug.log file is going more than 12M bcz of compilation and installation logs getting captured. Moreover Network test bucket require LTP to be installed in host and peer lpars. So i am downloading ltp in /home/ltp path and deleting the directory at the end of each case.
> > > Would you please let me know should i go to the old file and modify the code for Network tests. I still have the old PR open i will modify there and resent it again.
> > 
> > 
> > @TasmiyaNalatwad nice catch, can you please send a fix to delete old compile files as this also hits us when we want to copy logs, I still advocate having it in single place, let us wait to hear from @abdhaleegit what he says
> 
> @PraveenPenguin Our initial plan and old PR we had in the same file, but by looking at the upcoming huge design change with LTP network to modify ltp binaries in local and remote host.. so I plan to keep this separate .. else the test will become complex for further enhancenement. so will keep ltp_network separate

@abdhaleegit I need help to  understand this more on let us have discussion offline

**Comment by abdhaleegit** [comment]
Closing this as the code is added into existing ltp test

**Pull request closed without merging**